packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -1299,13 +1299,86 @@ class RenderFractionalTranslation extends RenderProxyBox {
   }
 }
 
+/// The interface used by [CustomPaint] (in the widgets library) and
+/// [RenderCustomPaint] (in the rendering library).
+///
+/// To implement a custom painter, subclass this interface to define your custom
+/// paint delegate.
+///
+/// The [paint] method is called whenever the custom object needs to be
+/// repainted. This is the primary method that must be implemented.
```

Comment:
The problem with the active voice in documentation like this is that it says more than it should. For example, the paint method might not be called by the framework, it might be called by code written by the developer that happens to use the same class. Similarly, it's not just the developer's subclass that has to implement the methods; suppose someone is reading the code to learn how the class works so that they can review someone else's code.

However, I've reworded this a bit to use your structure.

I've removed the mention of "delegate class".